hera_sim/interpolators.py

```diff
@@ -0,0 +1,99 @@
+"""
+This module provides interfaces to different interpolation classes.
+"""
+
+import numpy as np
+from scipy.interpolate import RectBivariateSpline
+
+class Tsky:
+    """
+    # TODO: fill in docstring
+    """
+    def __init__(self, datafile, **interp_kwargs):
+        self.datafile = datafile
+        self._data = np.load(self.datafile, allow_pickle=True)
+        # make sure the correct keys are used
+        assert 'freqs' in self._data.keys(), \
+                "The frequencies corresponding to the sky temperature array " \
+                "must be provided. They must be saved to the npz file using " \
+                "the key 'freqs'."
+        assert 'lsts' in self._data.keys(), \
+                "The LSTs corresponding to the sky temperature array must " \
+                "be provided. They must be saved to the npz file using the " \
+                "key 'lsts'."
+        assert 'tsky' in self._data.keys(), \
+                "The sky temperature array must be saved to the npz file " \
+                "using the key 'tsky'."
+        assert 'meta' in self._data.keys(), \
+                "The npz file must contain a metadata dictionary that can " \
+                "be accessed with the key 'meta'. This dictionary should " \
+                "provide information about the units of the various arrays " \
+                "and the polarizations of the sky temperature array."
+        # now make sure that the sky temperature array has the correct shape
+        # this should be (NPOLS, NLSTS, NFREQS)
+        # XXX this is somewhat strict, but useful in the case NFREQS==NLSTS
+        # XXX however, in the case NFREQS==NLSTS, it is possible for the
+        # XXX sky temperature to be shaped incorrectly and pass this assertion
+        assert self.tsky.shape==(len(self.meta['pols']),
+                                 self.lsts.size, self.freqs.size), \
+                "The tsky array is incorrectly shaped. Please ensure that " \
+                "the tsky array has shape (NPOLS, NLSTS, NFREQS)."
+
+        self._interpolators = {}
+
+        # fill in interpolators
+        # XXX do we want to make a hidden member function for generating
```

Comment:
I think having hidden member properties (actually, `cached_property`s) is a better way to go. This class should be considered a `view` into a specific `npz` file, and should disallow changing the data. There's no reason to ever change the data, which makes it _more clear_ what's going on by making them properties or cached properties.